titan-compiler/util.lua

```
@@ -1,10 +1,17 @@
 local util = {}
 
 function util.get_file_contents(filename)
-    local f = assert(io.open(filename, "r"))
+    local f, err = io.open(filename, "r")
+    if not f then
+        return false, err
+    end
     local s = f:read("a")
     f:close()
-    return s
+    if not s then
+        return false, "file '" .. filename .. "' is empty"
+    else
```

Do we really need to produce an error on empty files?

---

If we don't produce an error, i think we should at least return ''.
What do you think?

---

Yes, I think returning the empty file would be the best case here. I think read("a") already does that though, without needing to treat it separately.

---

That's true, but on my system, f:read returns nil if the file is a directory.
`lua -e "f = io.open('.'); print(type(f:read('a')))"`
I think we should return a generic error message in this case.